test/unit/caching/test_caching.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Ericsson AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Functional test, to check if caching is working correctly
+"""
+import unittest
+import os
+from typing import final
+from common.base import TestBase
+
+
+class TestCaching(TestBase):
+    """Caching tests"""
+
+    # Set working directory
+    __test_path__ = os.path.dirname(os.path.abspath(__file__))
+    BAZEL_BIN_DIR = os.path.join(
+        "../../..", "bazel-bin", "test", "unit", "caching"
+    )
+    BAZEL_TESTLOGS_DIR = os.path.join(
+        "../../..", "bazel-testlogs", "test", "unit", "caching"
+    )
+
+    @final
+    @classmethod
+    def setUpClass(cls):
+        """Clean up before the test suite"""
+        super().setUpClass()
+        cls.run_command("bazel clean")
+
+    def setUp(self):
+        """Before every test: clean Bazel cache"""
+        super().setUp()
+        self.run_command("mkdir tmp")
+        self.run_command("cp primary.cc secondary.cc linking.h BUILD tmp/")
+
+    def tearDown(self):
+        """Clean up working directory after every test"""
+        super().tearDown()
+        self.run_command("rm -rf tmp")
```

Comment:
Prefer standard python library solutions (shutil, os, pathlib) to system calls. See CodeChecker: https://github.com/Ericsson/codechecker/blob/master/analyzer/tests/functional/analyze/test_analyze.py#L49